src/Workspaces/Core/Portable/Options/OptionServiceFactory.cs

```diff
@@ -121,7 +123,12 @@ public OptionSet GetOptions()
             public T GetOption<T>(Option<T> option) => _globalOptionService.GetOption(option);
             public T GetOption<T>(PerLanguageOption<T> option, string languageName) => _globalOptionService.GetOption(option, languageName);
             public IEnumerable<IOption> GetRegisteredOptions() => _globalOptionService.GetRegisteredOptions();
-            public void SetOptions(OptionSet optionSet) => _globalOptionService.SetOptions(optionSet);
+
+            public void SetOptions(OptionSet optionSet)
+            {
+                _globalOptionService.SetOptions(optionSet);
+                _workspace.OnOptionsChanged();
```

Comment:
hmmm... this is strange. we no longer raise http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Options/IOptionService.cs,50 ?